src/optix_api.cpp

```diff
@@ -810,17 +811,25 @@ void jitc_optix_launch(ThreadState *ts, const Kernel &kernel,
         sbt.callablesRecordCount = kernel.optix.pg_count - 1;
     }
 
-    if (launch_size >= 0x40000000u)
-        jitc_raise("jit_optix_launch(): attempted to launch a very large "
-                   "wavefront of size %u. The maximum wavefront size permitted "
-                   "by OptiX is 2**30 == 1073741824. Please render using "
-                   "multiple passes, use fewer samples, or a lower resolution.",
-                   launch_size);
```

Comment:
Don't we still need a error message when the wavefront size is greater than 2^32?

Reply:
You cannot create Dr.Jit arrays wit more than 2^32 entries. The `jit_var_new*` variables will generate an error message. That error message is arguably a little bit low level in the rendering context, and that's why the Mitsuba part also has a more detailed message relating this to the image resolution and number of spp.